test/gtest/uct/test_p2p_am.cc

```diff
@@ -204,13 +204,25 @@ class uct_p2p_am_test : public uct_p2p_test
         }
     }
 
-    virtual void test_xfer(send_func_t send, size_t length, direction_t direction) {
+    virtual int is_op_supported_with_mem_type(send_func_t send,
+                                              uct_memory_type_t mem_type) {
+        if (mem_type == UCT_MD_MEM_TYPE_CUDA) {
+            if ((send == &uct_p2p_am_test::am_short) ||
```

Comment:
why is that needed? AFAIK cuda TLs do not support AM anyway?

Reply:
ok, i got it now. so maybe, instead of `direction`, pass down `flags`?
and the test will run cuda mem if both sides are zcopy
later on, we will add more flags which would allow also testing am_zcopy
```
TEST_UCT_FLAG_DIR_SEND_TO_RECV = UCS_BIT(0),
TEST_UCT_FLAG_SEND_ZCOPY = UCS_BIT(1),
TEST_UCT_FLAG_RECV_ZCOPY = UCS_BIT(2),
```

Reply:
Fixed. please review